internal/datastore/postgres/caveat.go

```diff
@@ -46,26 +46,32 @@ func (r *pgReader) ReadCaveatByName(ctx context.Context, name string) (*core.Cav
 	}
 	defer txCleanup(ctx)
 
-	var expr []byte
+	var serializedDef []byte
 	var rev datastore.Revision
-	err = tx.QueryRow(ctx, sql, args...).Scan(&expr, &rev)
+	err = tx.QueryRow(ctx, sql, args...).Scan(&serializedDef, &rev)
 	if err != nil {
 		if errors.Is(err, pgx.ErrNoRows) {
 			return nil, datastore.NoRevision, datastore.NewCaveatNameNotFoundErr(name)
 		}
 		return nil, datastore.NoRevision, err
 	}
-	return &core.CaveatDefinition{
-		Name:                 name,
-		SerializedExpression: expr,
-	}, rev, nil
+	def := core.CaveatDefinition{}
+	err = def.UnmarshalVT(serializedDef)
+	return &def, rev, err
 }
 
-func (r *pgReader) ListCaveats(ctx context.Context) ([]*core.CaveatDefinition, error) {
-	ctx, span := tracer.Start(ctx, "ListCaveats")
+func (r *pgReader) ListCaveats(ctx context.Context, caveatNames ...string) ([]*core.CaveatDefinition, error) {
+	ctx, span := tracer.Start(ctx, "ListCaveats", trace.WithAttributes(
+		attribute.StringSlice("names", caveatNames),
+	))
 	defer span.End()
 
-	filteredListCaveat := r.filterer(listCaveat)
+	caveatsWithNames := listCaveat
+	if len(caveatNames) > 0 {
```

Comment:
I wonder if we should `nil` check here and if not nil and empty, just return nothing. @jzelinskie Thoughts?

Reply:
Checked: it is empty by default, so this is fine

Reply:
We'll just need to be extra careful with the call sites